docs/ondiskformat.md

```diff
@@ -0,0 +1,168 @@
+# on-disk format
+
+Ursadb tries to keep its on-disk format simple. There are 7 types of files:
+
+- [`dataset`](#dataset) file (json)
+- [`database`](#database) file (json)
+- [`index`](#index) file (binary)
+- [`names`](#names) file (plaintext)
+- [`namecache`](#namecache) file (binary)
+- [`itermeta`](#itermeta) file (json)
+- [`iterator`](#iterator) file (plaintext)
+
+## database
+
+Contains references to all the other files in ursadb.
+If something is not (directly or indirectly) referenced from the database file,
+it means that it's not used and can be safely removed (when the database is turned off). 
+
+Example:
+
+```
+{
+    "config": {
+        "database_workers": 10
+    },
+    "datasets": [
+        "set.507718ac.db.ursa"
+    ],
+    "iterators": {
+        "01ef4552": "itermeta.01ef4552.db.ursa",
+    },
+    "version": "1.3.2"
+}
+```
+
+## dataset
+
+Contains all necesary information about a [dataset](./datasets.md).
```

Comment:
Not really enlightening. Maybe add some explanations about fields in the file?

Reply:
Hard to say something interesting that's not also explained by the example. Still tried tho